lib/omniauth/strategies/google-oauth2-access-token.rb

```diff
@@ -81,6 +81,10 @@ def request_phase
       end
 
       def callback_phase
+        if !request.params['auth_code'].to_s.empty? && request.params['access_token'].to_s.empty?
```

Comment:
that first half would be more clear if it wasn't a dobule negative:
`request.params['auth_code'].to_s.present?`